src/hangar/dataset/torch.py

```diff
@@ -0,0 +1,98 @@
+from typing import Sequence, Callable, TYPE_CHECKING
+
+from torch.utils.data.dataset import Dataset
+from .common import Dataset as HangarDataset
+from ..utils import experimental
+
+if TYPE_CHECKING:
+    from hangar.columns.column import ModifierTypes as Columns
+
+
+class TorchDataset(Dataset):
```

Comment:
I would spell this `class TorchDataset(torch.utils.data.dataset.Dataset):` to make it more clear which `Dataset` it's inheriting from.

Likewise, I'd rename `.common.Dataset` to `.common.HangarDataset` globally.